src/Microsoft.Health.Fhir.Shared.Api/Modules/HealthChecks/CachedHealthCheck.cs

```diff
@@ -44,7 +44,16 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 return _lastResult;
             }
 
-            await _semaphore.WaitAsync(cancellationToken);
+            try
+            {
+                await _semaphore.WaitAsync(cancellationToken);
+            }
+            catch (OperationCanceledException oce) when (cancellationToken.IsCancellationRequested)
+            {
+                _logger.LogInformation(oce, $"Cancellation was requested for {nameof(CheckHealthAsync)}");
+                return HealthCheckResult.Healthy("Health check was canceled before completion.");
```

Comment:
Open to suggestions on a better status response to return.

Reply:
Likely, if the request has been cancelled the client is not waiting for the response

Reply:
Yes, it shouldn't matter what we return since the caller is not going to get the response. But we need to return something. Maybe I can just return _lastResult (which will either be assigned or default values)

Reply:
I think returning last_Result makes sense.

Reply:
Updated to return _lastResult